deeppavlov/dataset_readers/mltrack_reader.py

```diff
@@ -0,0 +1,247 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, softwaredata
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+import re
+from typing import Dict, List, Tuple
+
+from deeppavlov.core.commands.utils import expand_path
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.data.dataset_reader import DatasetReader
+
+
+@register("mltrack_reader")
+class MLtrackReader(DatasetReader):
+    """The class to read the MLtrack dataset from files.
+
+    Please, see https://contest.yandex.ru/algorithm2018
+    """
+
+    def read(self,
+             data_path: str,
+             seed: int = None, *args, **kwargs) -> Dict[str, List[Tuple[List[str], int]]]:
+        """Read the MLtrack dataset from files.
+
+        Args:
+            data_path: A path to a folder with dataset files.
+            seed: Random seed.
+        """
+
+        data_path = expand_path(data_path)
+        train_fname = data_path / "train.tsv"
+        test_fname = data_path / "final.tsv"
+        train_data = self.build_train_data(train_fname)
+        valid_data = self.build_validation_data(train_fname, "valid")
+        test_data = self.build_validation_data(test_fname, "test")
+        dataset = {"train": train_data, "valid": valid_data, "test": test_data}
+        return dataset
+
+    def build_validation_data(self, name, data_type):
+        lines = []
+        with open(name, "r") as f:
+            reader = csv.reader(f, delimiter="\t", quotechar=None)
+            for line in reader:
+                lines.append(line)
```

Comment:
Maybe something like `lines = list(reader)`?

Reply:
Now your version is used